Solution1/Solution1.Tests/BL/Services/WeatherServiceTests.cs

```diff
@@ -40,8 +59,69 @@ public async Task GetByCityNameAsync_ReturnedWeatherDTO_Success()
             var result = await _weatherService.GetByCityNameAsync(cityName);
 
             // Assert
-            var expectedWeatherDto = new WeatherDTO() { CityName = cityName, Temp = temp, Comment = "It's fresh." };
+            var expectedWeatherDto = new WeatherDTO() { CityName = cityName, Temp = temp, Comment = comment };
+            Assert.True(new CompareLogic().Compare(expectedWeatherDto, result).AreEqual);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(3)]
+        public async Task GetForecastByCityNameAsync_ReturnedForecastWeatherDTO_Success(int countDays)
+        {
+            // Arrange
+            var startForecast = new DateTime(2022, 10, 12, 00, 00, 00);
+            var weatherPointPeriod = 3;
+            var countWeatherPointsinDay = 24/ weatherPointPeriod;
+            var countWeatherPoint = countWeatherPointsinDay * countDays + (DateTime.UtcNow.Date.AddDays(1) - DateTime.UtcNow).Hours / weatherPointPeriod;
+            var forecastWeatherApiDTO = new ForecastWeatherApiDTO()
+            {
+                City = new CityApiDTO() { Name = cityName },
+                WeatherPoints = new List<WeatherInfoApiDTO>()
+            };
+            var validationResult = new ValidationResult(new List<ValidationFailure>());            
+
+            for (int currentCountDays = 0; currentCountDays < countDays; currentCountDays++)
+            {
+                for(int currentHours = 0; currentHours < 24; currentHours += weatherPointPeriod)
+                {
+                    forecastWeatherApiDTO.WeatherPoints
+                        .Add(
+                            new WeatherInfoApiDTO()
+                            {
+                                DateTime = startForecast.AddDays(currentCountDays).AddHours(currentHours),
+                                Temp = new TempApiDTO() { Value = temp }
+                            });
+                }
+            }
+
+            _validator
+                .Setup(validator => validator.ValidateAsync
+                    (It.IsAny<ValidationContext<ForecastWeatherRequestDTO>>(),
```

Comment:
here It.Is should be used, to limit for by city name and days count

Reply:
+